registrations/tests.py

```diff
@@ -3314,3 +3314,101 @@ def test_generate_and_send_invalid_json(
             'foo.bar', datetime.utcfromtimestamp(300),
             datetime.utcfromtimestamp(500))
         mock_send_metric.assert_not_called()
+
+
+def override_get_data(self):
+    return [{
+        "mothers_phone_number": "07031221927",
+        "pregnancy_week": "13",
+        "preferred_msg_language": "english",
+        "type_of_registration": "prebirth",
+        "preferred_msg_type": "voice",
+        "message_days": "tuesday_and_thursday",
+        "message_time": "2-5pm",
+        "message_receiver": "mother_and_father",
+        "gatekeeper_phone_number": "07031221927"
+    }]
+
+
+def override_get_data_bad(self):
+    return [{
+        "mothers_phone_number": "",
+        "pregnancy_week": "13",
+        "preferred_msg_language": "english",
+        "type_of_registration": "prebirth",
+        "preferred_msg_type": "voice",
+        "message_days": "tuesday_and_thursday",
+        "message_time": "2-5pm",
+        "message_receiver": "",
+        "gatekeeper_phone_number": ""
+    }]
+
+
+class TestThirdPartyRegistrations(AuthenticatedAPITestCase):
+
+    def setUp(self):
+        tasks.PullThirdPartyRegistrations.get_data = override_get_data
```

Comment:
This is quite a large function, that does a lot. I think we should have tests that cover it. I don't think that these existing tests need to cover the `get_data` function, but I think there should be additional tests covering the functionality of the `get_data` function.